modules/andes-core/client/src/main/java/org/wso2/andes/jndi/Utils.java

```diff
@@ -61,6 +63,11 @@ public static String resolveSystemProperty(String text) {
 
         int indexOfStartingChars = -1;
         int indexOfClosingBrace;
+        
+        // Temporarily replace the secure vault alias ($secret{alias}) with a placeholder
+        String secAlias = StringUtils.substringBetween(text, SEC_PREFIX, PLACEHOLDER_SUFFIX);
```

Comment:
is this capture immediate PLACEHOLDER_SUFFIX

Reply:
no this capture the PLACEHOLDER_SUFFIX after `$sec{` . ex: for `amqp://${carbon.admin}:$secret{admin_password}@clientid/carbon?brokerlist='tcp://${carbon.local.ip}:${jms.port}'` type of urls work, `admin_password` is retrieved as secAlias